utils/clean_txt.py

```diff
@@ -2,59 +2,108 @@
 """A command line utility to clean text files generated from PDFs for Praxis-NLP."""
 import argparse
 from pathlib import Path
-from sys import stderr
 import re
 
-BRACKET_REGEX = re.compile(r"\s?\[\w*\]\s?|\s?\(\w*\)\s?")
-BIB_REGEX = re.compile(r"(Bibliography)|(Works Cited)|(Source Extracts)", re.IGNORECASE)
+# match expressions in square or round brackets containing alphanumeric
+# characters or spaces
+BRACKET_REGEX = re.compile(r'\s?\[(\w|\s|,)*\]|\s?\((\w|\s|,)*\)')
+BIB_REGEX = re.compile(
+    r'(Bibliography)|(Works Cited)|(Source Extracts)|(Appendix(:|\n))', re.IGNORECASE
+)
 EXTRA_WHITESPACE = re.compile(r" [a-z\)\],;]+\n")
 
 
+def remove_non_ascii(s):
+    """Re-encodes s as ASCII"""
+    s.encode('ascii', 'ignore')
+    return s
+
+
 def remove_citations(s, verbose=False):
-    """Removes anything between brackets, parentheses, or after a works cited heading"""
-    lines=[]
+    """Removes anything between brackets, or after a bibliograhpy heading"""
+    lines = []
     characters_written = 0
     characters_read = 0
 
     for line_number, line in enumerate(s.split('\n')):
 
         if re.search(BIB_REGEX, line):
             if verbose:
-                print(f"Truncated at line {line_number}: {line}", file=stderr)
+                print(f'Truncated at line {line_number}: {line}')
             break
-        (new_line, __) = re.subn(BRACKET_REGEX, "", line)
+        (new_line, __) = re.subn(BRACKET_REGEX, '', line)
         lines.append(new_line)
         characters_read += len(line)
         characters_written += len(new_line)
 
-    characters_read += sum(len(line) for line in s)
```

Comment:
How does characters read stay accurate here without this statement. The original goal was to make sure we weren't deleting like 90% of the file after the bibliography regex matched, now it will only account for characters removed as citations. Is that the new intention?

Reply:
Fair, although how it was previously was also wrong - we were adding to characters to `characters_read` in two places, which lead to erroneous deletion warnings all over the shop when I actually added that in.

Reply:
You're completely right, the way I was meaning to be doing it got absolutely lost in the shuffle and ended up being nonsensical. Didn't catch that first time around.

Reply:
Resolved with new commit.